addons/hr_holidays/report/hr_leave_employee_type_report.py

```diff
@@ -53,14 +53,30 @@ def init(self):
                     allocation.employee_id as employee_id,
                     employee.active as active_employee,
                     CASE
-                        WHEN allocation.id = min_allocation_id.min_id
-                            THEN aggregate_allocation.number_of_days - COALESCE(aggregate_leave.number_of_days, 0)
-                            ELSE 0
+                        WHEN allocation.state = 'validate' THEN
+                            allocation.number_of_days
+                                - COALESCE((SELECT SUM(l.number_of_days)
+                                            FROM hr_leave l
+                                            WHERE l.employee_id = allocation.employee_id
+                                              AND l.holiday_status_id = allocation.holiday_status_id
+                                              AND l.state IN ('validate', 'validate1')
+                                              AND l.date_from >= allocation.date_from
+                                              AND (allocation.date_to IS NULL
+                                               OR l.date_from <= allocation.date_to)), 0)
+                        ELSE 0
```

Comment:
Any reason not to use a condition to exclude not-approved allocations?
Also why include leaves in validate1 state? Those aren't fully validated yet :slightly_smiling_face:
Last but not least, this is wrong. Overlapping allocations are making the computation of links between allocations/leaves more complicated than just linking them by dates, hence the huge method `_get_consumed_leaves` :thinking: In my opinion, the option to choose a period for the balance should just be dropped